trestle/core/commands/author/jinja.py

```diff
@@ -158,10 +166,33 @@ def jinja_ify(
                 new_output = template.render(**lut)
 
             output_file = trestle_root / r_output_file
-            output_file.open('w', encoding=const.FILE_ENCODING).write(output)
+            if number_captions:
+                output_file.open('w', encoding=const.FILE_ENCODING).write(_number_captions(output))
+            else:
+                output_file.open('w', encoding=const.FILE_ENCODING).write(output)
 
         except Exception as e:  # pragma: no cover
             logger.error(f'Unknown exception {str(e)} occured.')
             logger.debug(traceback.format_exc())
             return 1
         return 0
+    
+def _number_captions(md_body) -> str:
+    """Incrementally number tables and image captions"""
+    images = {}
+    tables = {}
+    output = md_body.splitlines()
+    
+    for n, line in enumerate(output):
+        if re.match('^!\[.+\]\(.+\)', line):
```

Comment:
Can the image be not at the beginning of the line?

Reply:
it probably always should be at the start of a line, but lets not punish authors who write badly formatted markdown